packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -67,6 +68,76 @@ describe('<DataGrid /> - Pagination', () => {
       }, 50);
     });
 
+    it('should trigger onPageChange once if page prop is set', () => {
+      const onPageChange = spy();
+
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} page={1} onPageChange={onPageChange} pageSize={1} />
+        </div>,
+      );
+      expect(onPageChange.callCount).to.equal(1);
```

Comment:
```suggestion
      expect(onPageChange.callCount).to.equal(0);
```

Reply:
> I have commented on my understanding of how it should behave. The mental model I have always seen for the change event is: "does something try to change the current state?". For instance, when doing server-side rendering, the default page is already 1, I don't see why it should fire.

Well, we can argue that when the grid initially loads, then it goes from page 0 to page 1. By firing this event on page 1 then developers would just load page 1 as any page using their onPageChange event, so they don't need to initialize the first page separately and use onPageChange for page 2.

Reply:
Do you have more details on page 0? What does this state mean? Is it valid? Why doesn't it alway start on page 1?

Reply:
> so they don't need to initialize the first page separately and use onPageChange for page 2.

This logic doesn't work server-side.

Reply:
On a related note, I think that we should consider changing the index for the default page. The Pagination component, which is meant for SEO starts at page 1. The TablePagination component which is meant for data grid use cases starts at page 0: https://material-ui.com/components/pagination/#table-pagination. It looks like DataGrid starts at page 1: https://material-ui.com/components/data-grid/pagination/#controlled-pagination. It's inconsistent.

Reply:
Can we create a separate issue for that pls

Reply:
Sure, done